### PR TITLE
Fix azure config

### PR DESCRIPTION
**Pull request by yunzheng1112**
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by yunzheng1112** [comment]
I'm trying to use AutoGPT in my Windows machine with Azure OpenAI resource, and found there are 3 issues:
1. azure-api-type should be "azure" instead of "azure-ad";
2. Azure config file location is incorrect;
3. Completion API call is invalid because of the wrong parameter name;
Fix them and it works.

**Review by fatmelon — APPROVED** [review]

**Comment by k-boikov** [comment]
Partially duplicate of https://github.com/Significant-Gravitas/Auto-GPT/pull/2214

**Comment by sweetlilmre** [comment]
@yunzheng1112 

> I'm trying to use AutoGPT in my Windows machine with Azure OpenAI resource, and found there are 3 issues:
> 
>     1. azure-api-type should be "azure" instead of "azure-ad";
> 
>     2. Azure config file location is incorrect;
> 
>     3. Completion API call is invalid because of the wrong parameter name;
>        Fix them and it works.

1. I think exposing `azure-api-type` is redundant, this shouldn't have been hoisted into config in the first place.
2. 100%
3. `deployment_id` works and is according to the documentation. Where do you see that it should be `engine`? (it is `engine` for the embedding call) see the docs linked in the README.md

**Comment by yunzheng1112** [comment]
> @yunzheng1112
> 
> > I'm trying to use AutoGPT in my Windows machine with Azure OpenAI resource, and found there are 3 issues:
> > ```
> > 1. azure-api-type should be "azure" instead of "azure-ad";
> > 
> > 2. Azure config file location is incorrect;
> > 
> > 3. Completion API call is invalid because of the wrong parameter name;
> >    Fix them and it works.
> > ```
> 
> 1. I think exposing `azure-api-type` is redundant, this shouldn't have been hoisted into config in the first place.
> 2. 100%
> 3. `deployment_id` works and is according to the documentation. Where do you see that it should be `engine`? (it is `engine` for the embedding call) see the docs linked in the README.md

The Completion example in this doc shows it should be "engine":
https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/chatgpt?pivots=programming-language-chat-completions

File: azure.yaml.template
Comment (nponeccop) [inline azure.yaml.template:1]: Is this related to the config issue? If no, revert.

File: autogpt/llm_utils.py
Comment (nponeccop) [inline autogpt/llm_utils.py:1]: This one looks like a separate fix

**Review by nponeccop — APPROVED** [review]

File: autogpt/llm_utils.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def create_chat_completion(
         try:
             if CFG.use_azure:
                 response = openai.ChatCompletion.create(
-                    deployment_id=CFG.get_azure_deployment_id_for_model(model),
+                    engine=CFG.get_azure_deployment_id_for_model(model),
```
Comment (Pwuts) [inline autogpt/llm_utils.py]: MS docs say `engine` https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/chatgpt?pivots=programming-language-chat-completions

OpenAI docs say `deployment_id` https://github.com/openai/openai-cookbook/blob/main/examples/azure/chat.ipynb

EDIT: These other MS docs *also* say `deployment_id`? https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#chat-completions

EDITEDIT: Swagger docs of the Azure OpenAI service say `deployment_id` https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-03-15-preview/inference.json

???

**Review by Pwuts — APPROVED** [review]

**Review by Pwuts — APPROVED** [review]
